src/steps/containers.rs

```diff
@@ -18,15 +19,42 @@ use crate::{execution_context::ExecutionContext, utils::require};
 // themselves or when using docker-compose.
 const NONEXISTENT_REPO: &str = "repository does not exist";
 
+/// Uniquely identifies a `Container`.
+#[derive(Debug)]
+struct Container {

```

Comment:
This type technically should be named `Image`,  but let's call it `Container` as this is what `topgrade` currently uses